phenol-annotations/src/main/java/org/monarchinitiative/phenol/annotations/io/hpo/HpoaDiseaseDataLoaderDefault.java

```diff
@@ -36,30 +34,40 @@ public HpoaDiseaseDataContainer loadDiseaseData(InputStream is) throws IOExcepti
 
     Map<TermId, List<HpoAnnotationLine>> annotationLinesByDiseaseId = new HashMap<>();
     boolean header = false;
-    HpoAnnotationMetadata metadata = new HpoAnnotationMetadata();
-    StringBuilder metaDataBuilder = new StringBuilder();
+    String version = null;
+    String hpoVersion = null;
     for (String line = reader.readLine(); line != null; line = reader.readLine()) {
       // Comment Lines
       if (line.startsWith("#")) {
-        metaDataBuilder.append(String.format("%s\n", line.replace("#", "")));
+        if(line.startsWith("#version:") || line.startsWith("#date:")){
+          Matcher matcher = VERSION_PATTERN.matcher(line);
+          if (matcher.matches()){
+            version = matcher.group("version");
+          }
+        } else if(line.toLowerCase().startsWith("#hpo-version:")){
+            Matcher matcher = HPO_VERSION_PATTERN.matcher(line);
+            if (matcher.matches()){
+              hpoVersion = matcher.group("hpoversion");
+            }
+        }
         continue;
       }
 
       // Header Line
-      // Build our yml for metadata to HpoaMetadata
       if (!header) {
-        Constructor c = new Constructor(HpoAnnotationMetadata.class, new LoaderOptions());
-        c.setPropertyUtils(new PropertyUtils() {
-          @Override
-          public Property getProperty(Class<? extends Object> type, String name){
-            name = CaseUtils.toCamelCase(String.join(" ", name.split("\\-")), false);
-            return super.getProperty(type, name);
-          }
-        });
-        final Yaml yaml = new Yaml(c);
-        metadata = yaml.load(metaDataBuilder.toString());
+        if (version == null || (!version.equals(hpoVersion))){
+          LOGGER.error("HPOA disease data version was not found or ontology mismatch.");
+          throw new IOException("HPOA disease data version was not found or ontology mismatch.");
+        }
         header = true;
-        continue;
+        // Old format catcher
```

Comment:
This comment seems to be out of place since the old format should not have a header line that starts with `database_id`. All header lines in the old format start with `#`. I'll remove the comment